### ISSUE TITLE
Prevent generate from failing on undefined imports

### ISSUE THREAD

**Issue by Danlock** [jerbob92/wazero-emscripten-embind#12]
Using go generate with Tesseract compiled to WASM leads to some errors complaining that some host imports are missing, such as system.

```
2023/10/27 22:53:23 "system" is not exported in module "env"
exit status 1
```

However, I intend to supply these host imports myself via wazero HostFunctionBuilder later on. Is there a way for me to tell the generator to ignore undefined host imports?

**Comment by jerbob92** [comment]
Hi! The generator needs the host methods implemented, because the generator needs the WASM to run to be able to generate the bindings. I did not really think of this issue though, can you post the WASM file here so I can work on a solution? My idea is to generate dummy host methods for these methods so that it can at least run the `_initialize` method to have Emscripten call all the embind register host methods.

**Comment by Danlock** [comment]
Yes, something like a comma separated list of module.function names would make sense, since the generator shouldn't need to have those working.

I feel like this is something that would be useful in wazero itself. Instead of it failing to compile WASM modules missing imports, a config option that compiles WASM but logs/panics when unspecified imports are called would be useful.
Here is the attached tesseract WASM file.
[tesseract-files.zip](https://github.com/jerbob92/wazero-emscripten-embind/files/13207230/tesseract-files.zip)

**Comment by jerbob92** [comment]
There are some things in there that should not be in a non-web build, that it generated a JS file is also a signal. Emscripten standalone is missing quite some things to support WASI, I have made this PR to add a lot of WASI to Emscripten: https://github.com/emscripten-core/emscripten/pull/18285

Can you show me your build command? Your issue is still legit but I still think you will have other issues after generation.

**Comment by Danlock** [comment]
I believe the problem with the host imports arise from Leptonica (Tesseract's image library) containing functions for reading images directly from the filesystem.

Note that I have already managed to get Tesseract working reading images using Emscripten's WebIDL generation and carefully calling the Emscripten 3.1.44 generated functions, but the code was ugly so I figured I would try your lib instead. I don't actually need any extra WASI functionality.

The previously attached WASM is a separate compile of Tesseract with Embind, instad of WebIDL to use this library, but other than failing on file I/O I hope nothing else breaks.

Also, I am intentionally outputting javascript alongside the WASM, as I used it as a reference to understand how Emscripten implements the host imports in javascript. 

Here are my emcc flags:

```
EMCC_FLAGS =\
  -O0\
  -g3\
  -sSTANDALONE_WASM\
  -sPURE_WASI\
  -sEXPORTED_FUNCTIONS="_malloc,_free"\
  --minify 0\
-sUSE_GIFLIB=1 \
-sUSE_ZLIB=1 \
-sUSE_LIBPNG=1 \
-sUSE_LIBJPEG=1\
  --no-entry\
  -sFILESYSTEM=0 \
  -sALLOW_MEMORY_GROWTH\
  -sMAXIMUM_MEMORY=1GB \
  -lembind \
  -std=c++20 \
  -fexperimental-library
  ```

**Comment by jerbob92** [comment]
I have created a fix in #13, could you check if that works for you? You will need to implement (can be a dummy method) the syscall methods to make Wazero happy when you want to run the generated code. It would be nice if Wazero could ignore that with a configuration, that would be a good suggestion for that project :)
